pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1686,3 +1696,12 @@ func outboundTunnelListener(proxy *model.Proxy) *listener.Listener {
 	)
 	return buildConnectOriginateListener(baggage)
 }
+
+// conflictWithStaticListener checks whether the given port conflicts with static listener port
+// default is 15021 and 15090
+func conflictWithStaticListener(proxy *model.Proxy, port int) bool {
```

Comment:
It should only be HTTP I think? Its fine to have TCP, or even "auto"

Reply:
15090 and 15021 are noth wildcard listener, so i think envoy does allow even TCP listener with these ports

Reply:
You can have 0.0.0.0 binding and 1.1.1.1 on the same port. We cannot block TCP services or it will likely break existing users. Should only block the wildcard bindings.

Checking wildcard is probably more robust than the protocol, btw

Reply:
I canot remember we generate both wildcard and 1.1.1.1 listeners on the same port. kind of remember we merge them in to widcard 0.0.0.0 with different filterchains